File: TranslationApplication/app/src/main/java/com/example/translationapplication/http/VolleyManager.java
package com.example.translationapplication.http;

import android.content.Context;

import com.android.volley.RequestQueue;
import com.android.volley.toolbox.Volley;

public class VolleyManager {
    private RequestQueue rq;

    private VolleyManager() {

    }

    private static class Singleton {
        private static final VolleyManager INSTANCE = new VolleyManager();
    }

    public static VolleyManager getInstance() {
        return Singleton.INSTANCE;
    }

    public RequestQueue getRequestQueue(Context context) {
        if (rq == null) {
            rq = Volley.newRequestQueue(context);
        }
        return rq;
    }
}

File: Kotlin/app/src/main/java/com/bowoon/android/kotlin/VolleyManager.kt
package com.bowoon.android.kotlin

import android.content.Context
import com.android.volley.RequestQueue
import com.android.volley.toolbox.Volley

class VolleyManager {
    private var rq: RequestQueue? = null

    init {

    }

    companion object {
        private class Singleton {
            companion object {
                val INSTANCE: VolleyManager = VolleyManager()
            }
        }

        fun getInstance(): VolleyManager = Singleton.INSTANCE
    }

    fun getRequestQueue(): RequestQueue? {
        if (rq == null) {
            throw IllegalAccessException("Need Initialize Request Queue")
        }
        return rq
    }

    fun setRequestQueue(context: Context) {
        this.rq = Volley.newRequestQueue(context)
    }
}